numba/tests/test_unicode.py

```diff
@@ -491,6 +503,119 @@ def try_compile_bad_optional(*args):
+            ('a' + '\U00100304' * 100, ['a']),
+            ('\u0102' + '\U00100304' * 100, ['\u0102']),
+            ('a' * 100, ['\u0102', '\U00100304', '\u0102a', '\U00100304a']),
+            ('\u0102' * 100, ['\U00100304', '\U00100304\u0102']),
+            ('_a' + '\u0102' * 100, ['_a']),
+            ('_a' + '\U00100304' * 100, ['_a']),
+            ('_\u0102' + '\U00100304' * 100, ['_\u0102']),
+        ]
+        for s, subs in default_subs + cpython_subs:
+            for sub_str in subs:
+                msg = 'Results "{}".rfind("{}") must be equal'
+                self.assertEqual(pyfunc(s, sub_str), cfunc(s, sub_str),
+                                 msg=msg.format(s, sub_str))
+
+    def test_rfind_with_start_only(self):
+        pyfunc = rfind_with_start_only_usecase
+        cfunc = njit(pyfunc)
+
+        for s in UNICODE_EXAMPLES:
+            for sub_str in ['', 'xx', s[:-2], s[3:], s]:
+                for start in list(range(-20, 20)) + [None]:
+                    msg = 'Results "{}".rfind("{}", {}) must be equal'
+                    self.assertEqual(pyfunc(s, sub_str, start),
+                                     cfunc(s, sub_str, start),
+                                     msg=msg.format(s, sub_str, start))
+
+    def test_rfind_with_start_end(self):
+        pyfunc = rfind_with_start_end_usecase
+        cfunc = njit(pyfunc)
+
+        starts = list(range(-20, 20)) + [None]
+        ends = list(range(-20, 20)) + [None]
+        for s in UNICODE_EXAMPLES:
+            for sub_str in ['', 'xx', s[:-2], s[3:], s]:
+                for start, end in product(starts, ends):
+                    msg = 'Results of "{}".rfind("{}", {}, {}) must be equal'
+                    self.assertEqual(pyfunc(s, sub_str, start, end),
+                                     cfunc(s, sub_str, start, end),
+                                     msg=msg.format(s, sub_str, start, end))
+
+    def test_rfind_wrong_substr(self):
+        cfunc = njit(rfind_usecase)
+
+        for s in UNICODE_EXAMPLES:
+            for sub_str in [None, 1, False]:
+                with self.assertRaises(TypingError) as raises:
+                    cfunc(s, sub_str)
+                msg = 'must be {}'.format(types.UnicodeType)
+                self.assertIn(msg, str(raises.exception))
+
+    def test_rfind_wrong_start_end(self):
+        cfunc = njit(rfind_with_start_end_usecase)
+
+        for s in UNICODE_EXAMPLES:
+            for sub_str in ['', 'xx', s[:-2], s[3:], s]:
+                for start, end in product([0.1, False], [-0.1, True, None]):
+                    with self.assertRaises(TypingError) as raises:
+                        cfunc(s, sub_str, start, end)
+                    accepted_types = (types.Integer, int, types.NoneType)
+                    msg = 'must be {}'.format(accepted_types)
+                    self.assertIn(msg, str(raises.exception))
+
+    @unittest.skipIf(utils.MACHINE_BITS == 32, 'No support for 32-bit')
```

Comment:
Perhaps use `types.intp` for a platform width integer opposed to banning this test on 32bit?